libcnb-test/README.md

```diff
@@ -46,3 +46,27 @@ fn call_test_fixture_service(addr: std::net::SocketAddr, payload: &str) -> Resul
    unimplemented!()
 }
 ```
+
+## Configuration
+
+### Main buildpack binary
+
+A buildpack crate can contain multiple binaries. To determine which binary is the main buildpack binary, the convention is:
+1. If there is only one binary, use that.
+2. If there are multiple, use the one that has the same name as the crate.
+
+If this convention does not work for your buildpack, you can explicitly configure the main buildpack binary in `Cargo.toml`:
```

Comment:
What use-cases do we see for needing to configure `buildpack-target` via `Cargo.toml`? Unless there is something significant, should we wait until we need this, before adding support?

Reply:
This is necessary when you don't put your `buildpack_main!` in `main.rs`, or configure your binary name via `Cargo.toml`. I feel like we should have this escape hatch from the get-go, otherwise folks would be stuck in such a case.

Reply:
Yeah I figured those were the times where the option might be necessary, but that still doesn't answer the question of what use-cases would this be used for?

libcnb intends to be opinionated - I think it's fine for us to pick opinionated sensible defaults and wait until someone asks before adding additional customisation options - particularly for `libcnb-cargo` which is for a very specific cross-compiling/packaging workflow. The problem with adding options up front, is we'll never know if anyone actually needs them or if they could be removed in the future.

For this option in particular, I'm struggling to think of a time where someone might want to customise the buildpack binary name but yet still be using libcnb-cargo. ie: Either they are using libcnb-cargo since they want a fully packaged buildpack (in which case the binary name in the `target/` directory is moot, since libcnb-cargo will rename it to `build` anyway), or else are doing something custom and likely not using `libcnb-cargo`.

For the "the buildpack isn't the main crate binary" case, I'm also not sure this is ever likely to happen. If say someone wants to bundle multiple buildpacks into the same crate, libcnb-cargo doesn't support that anyway. Or if they say have some other CLI tool that they want as the main binary, with the buildpack as an additional binary, then libcnb-cargo is going to bundle that CLI tool into the package whether they like it or not. And if the CLI tool is meant to be packaged into the buildpack (because it's some utility or intended for exec.d etc), then why isn't it a secondary binary in the crate, given it's purpose is less important than the buildpack itself?

Reply:
@edmorley and I discussed this for a bit in Slack. We decided to remove the explicit config (dfc90e8) for the reasons outlined by Ed above. It can be easily added back in should we ever need it.